### PR TITLE
Search IDIR (Active Directory) for users

### PR DESCRIPTION
**Pull request by pbolduc**
# Description

This change implements searching IDIR via LDAP.  It requires configuration changes.  See the README.md file in the API folder.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: api/src/BcGov.Malt.Web/Services/LdapUserSearchService.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        private User MapSearchResult(SearchResultEntry entry)
+        {
+            var user = new User
+            {
+                Id = GetAttributeValue(entry, "bcgovGUID") ?? string.Empty,
+                UserName = GetAttributeValue(entry, "sAMAccountName") ?? string.Empty,
+                FirstName = GetAttributeValue(entry, "givenName") ?? string.Empty,
+                LastName = GetAttributeValue(entry, "sn") ?? string.Empty,
+                Email = GetAttributeValue(entry, "mail") ?? string.Empty,
+            };
```
Comment (ChrisHoban) [inline api/src/BcGov.Malt.Web/Services/LdapUserSearchService.cs]: We don't map UPN Here?

**Comment by ChrisHoban** [comment]
Will also need to pull/resolve before merge can be complete

File: api/src/BcGov.Malt.Web/Services/LdapUserSearchService.cs (hunk 1 above)
Comment (pbolduc) [reply api/src/BcGov.Malt.Web/Services/LdapUserSearchService.cs]: UPN is not used by the front end.  I was going to address this when Nancy has the SharePoint stuff almost ready.

Comment (ChrisHoban) [reply api/src/BcGov.Malt.Web/Services/LdapUserSearchService.cs]: Gotcha, was wondering if it was something along those lines. Just need to unblock merging then due to readme.md conflicts

**Pull request merged**